patreon_archiver/main.py

```diff
@@ -69,8 +79,26 @@ def main(
         output_dir = Path('.', campaign_id)
     output_dir.mkdir(parents=True, exist_ok=True)
     chdir(output_dir)
+
+    session: requests.Session | None = None
+    if cookies_json is not None:
+        session = requests.Session()
+        session.headers.update(SHARED_HEADERS)
+        with open(cookies_json) as f:
+            cookies_data = json.load(f)
+        for cookie in cookies_data:
+            session.cookies.set(
+                cookie['name'],
+                cookie['value'],
+                domain=cookie.get('domain', '').lstrip('.'),
+                path=cookie.get('path', '/'),
+            )
```

Comment:
Please add a test covering the passing of `--cookies-json ...` in `test_main.py`.